sdk/containerregistry/azure-containerregistry/tests/perfstress_tests/list_artifacts.py

```diff
@@ -0,0 +1,35 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from azure_devtools.perfstress_tests import PerfStressTest
+from azure.containerregistry import ContainerRegistryClient
+from azure.containerregistry.aio import ContainerRegistryClient as AsyncContainerRegistryClient
+
+
+class ListArtifactsTest(PerfStressTest):
+
+    def __init__(self, arguments):
+        super().__init__(arguments)
+
+        # Create clients
+        account_url = self.get_from_env("CONTAINERREGISTRY_ANONREGISTRY_ENDPOINT")
+        audience = "https://management.azure.com"
+        self.anon_client = ContainerRegistryClient(endpoint=account_url, credential=None, audience=audience)
+        self.async_anon_client = AsyncContainerRegistryClient(endpoint=account_url, credential=None, audience=audience)       
+        self.repository = "node"
+
+    async def global_setup(self):
+        await super().global_setup()
+
+    async def close(self):
+        await self.async_anon_client.close()
+        await super().close()
+
+    def run_sync(self):
+        for manifest in self.anon_client.list_manifest_properties(self.repository):
+            self.anon_client.get_manifest_properties(self.repository, manifest.digest)
```

Comment:
This test seems to be combining multiple operations in a single perf-run.
We don't normally do that, as it makes it harder to see and debug where perf loss is happening.
Is there a reason you've done it here? Is this the scenario that's being tested across languages?

Reply:
Yes, I followed what they did in other languages, and both tests are being tested across languages. i.e. [JS](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/containerregistry/perf-tests/container-registry/test/listArtifacts.spec.ts#L15), [.NET](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/containerregistry/Azure.Containers.ContainerRegistry/perf/Azure.Containers.ContainerRegistry.Perf/Scenarios/ListArtifacts.cs#L45)

Reply:
JS is only calling service in the listing method, the for loop is only doing local operation `getArtifact()` to create client side helper.  Since Python is using a flatter hierarchy, `get_manifest_properties()` is likely different from other languages' `getArtifact()`

Reply:
Thanks @jeremymeng!
In that case @YalinLi0312 can you made sure that `get_manifest_properties` is either not making any service requests - or replace it with something that doesn't?
You can check the `ListBlobsTest` for an example - the just do an empty loop using `pass`

Reply:
By checking the src code, `get_manifest_properties` does making service requests([link](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/containerregistry/azure-containerregistry/azure/containerregistry/_generated/operations/_container_registry_operations.py#L1491)). And same in `get_repository_properties`. In this case, I think python should have 4 separate tests for each operation.